sdk/identity/identity/test/internal/node/managedIdentityCredential.spec.ts

```diff
@@ -360,18 +361,36 @@ describe("ManagedIdentityCredential", function () {
   it("sends an authorization request correctly in an Cloud Shell environment", async () => {
     // Trigger Cloud Shell behavior by setting environment variables
     process.env.MSI_ENDPOINT = "https://endpoint";
-
     const authDetails = await testContext.sendCredentialRequests({
       scopes: ["https://service/.default"],
-      credential: new ManagedIdentityCredential("client"),
+      credential: new ManagedIdentityCredential(),
       secureResponses: [createResponse(200, { access_token: "token" })],
     });
-
+    console.dir(authDetails);
     const authRequest = authDetails.requests[0];
     assert.equal(authRequest.method, "POST");
     assert.equal(authDetails.result!.token, "token");
   });
 
+  it("authorization request fails with client id passed in an Cloud Shell environment", async () => {
+    // Trigger Cloud Shell behavior by setting environment variables
+    process.env.MSI_ENDPOINT = "https://endpoint";
+    setLogLevel("warning");
+    const authDetails = await testContext.sendCredentialRequests({
+      scopes: ["https://service/.default"],
+      credential: new ManagedIdentityCredential("client"),
+      secureResponses: [createResponse(200, { access_token: "token" })],
+    });
+    console.dir(authDetails);
+    assert.equal(authDetails.result, null);
+    assert.equal(authDetails.error?.name, "CredentialUnavailableError");
+    assert.equal(
+      authDetails.error?.message,
+      "ManagedIdentityCredential: Authentication failed. Message No responses left."
+    );
+    assert.equal(authDetails.requests.length, 0);
```

Comment:
This test is identical to other tests that we have in this file, except for having enabled the logs through `setLogLevel`. Let’s try something fun: In the cloudShellMsi, export the logger, then here use Sinon to spy the logger and see if the `logger.warning` function has been invoked. How does that sound?